lib/puppet/feature/base.rb

```diff
@@ -77,6 +77,7 @@ def self.is_implemented
           true
         rescue LoadError => err
           Puppet.debug("CreateSymbolicLink is not available")
+          Puppet.deprecation_warning("Windows 2003 and below versions are deprecated, and will be officially unsupported in the next major Puppet release.  Please plan accordingly and see http://links.puppetlabs.com/deprecate-windows-2003 for more information.")
```

Comment:
It's own check makes more sense to me. E.g. I know nano drops some APIs (maybe not this one), and who knows what might come down the pike next. An explicit version check seems safest.

Also wording wise, some questions:
1) replace "Windows 2003 and below versions are deprecated" with "Support for Windows 2003 and below versions are deprecated"?
2) should we call out 2003r2 as well?
3) it says "next major puppet version". On the foss side does that really mean puppet 5? Or can it be puppet 4.3?